server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/event/PreComputeMessagePreviewListener.java

```diff
@@ -0,0 +1,102 @@
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.event;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.apache.james.jmap.api.preview.MessagePreviewStore;
+import org.apache.james.jmap.api.preview.Preview;
+import org.apache.james.jmap.draft.model.message.view.MessageFullViewFactory;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageIdManager;
+import org.apache.james.mailbox.events.Event;
+import org.apache.james.mailbox.events.Group;
+import org.apache.james.mailbox.events.MailboxListener;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.model.FetchGroupImpl;
+import org.apache.james.mailbox.model.MessageId;
+import org.apache.james.mailbox.model.MessageResult;
+import org.parboiled.common.ImmutableList;
+
+import com.github.fge.lambdas.Throwing;
+import com.github.steveash.guavate.Guavate;
+
+import reactor.core.publisher.Mono;
+
+public class PreComputeMessagePreviewListener implements MailboxListener.GroupMailboxListener {
+    public static class PreComputeMessagePreviewListenerGroup extends Group {
+
+    }
+
+    private static final Group GROUP = new PreComputeMessagePreviewListenerGroup();
+
+    private final MessageIdManager messageIdManager;
+    private final MessagePreviewStore messagePreviewStore;
+    private final MessageFullViewFactory messageFullViewFactory;
+
+    private MailboxManager mailboxManager;
+
+    @Inject
+    public PreComputeMessagePreviewListener(MailboxManager mailboxManager, MessageIdManager messageIdManager,
+                                            MessagePreviewStore messagePreviewStore, MessageFullViewFactory messageFullViewFactory) {
+        this.mailboxManager = mailboxManager;
+        this.messageIdManager = messageIdManager;
+        this.messagePreviewStore = messagePreviewStore;
+        this.messageFullViewFactory = messageFullViewFactory;
+    }
+
+    @Override
+    public Group getDefaultGroup() {
+        return GROUP;
+    }
+
+    @Override
+    public void event(Event event) throws MailboxException {
+        if (event instanceof Added) {
+            MailboxSession session = mailboxManager.createSystemSession(event.getUsername());
```

Comment:
Use SessionProvider instead.

Reply:
I tried to but I'm not convinced TBH... are you sure?

Reply:
Yes! We don't need the entire mailboxManager here, and I personnally would like to remove some responsabilities from that too complex component...